src/dawn/CodeGen/GridTools/GTCodeGen.cpp

```diff
@@ -566,6 +578,28 @@ GTCodeGen::generateStencilInstantiation(const StencilInstantiation* stencilInsta
 
     StencilConstructor.startBody();
 
+    // Add static asserts to check halos against extents
+    StencilConstructor.addComment("Check if extents do not exceed the halos");
+    for(int i = 0; i < numFields; ++i) {
+      if(!StencilFields[i].IsTemporary) {
+        auto max_extents = collectedExtents[StencilFields[i].Name];
```

Comment:
if you have a method 
``` std::unordered_map<int, Extents> Stencil::computeEnclosingAccessExtents()``` 
I would reference the field via accessID rather than name